### PR TITLE
[Locale] Current locale set on request

### PR DESCRIPTION
**Pull request by lchrusciel**
| Q               | A
| --------------- | -----
| Branch?         | 1.9
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

Note. PhpUnit tests were replaced with behat tests

### CODE REVIEW

File: src/Sylius/Bundle/CoreBundle/Resources/config/services/listeners.xml [hunk 1]
```diff
@@ -91,6 +91,10 @@
             <tag name="kernel.event_listener" event="sylius.address.pre_create" method="preCreate" />
         </service>
 
+        <service id="Sylius\Bundle\CoreBundle\EventListener\LocaleAwareListener" decorates="locale_aware_listener">
+            <argument type="service" id=".inner" />
```
Comment (lchrusciel) [inline src/Sylius/Bundle/CoreBundle/Resources/config/services/listeners.xml:95]:
```suggestion
            <argument type="service" id="Sylius\Bundle\CoreBundle\EventListener\LocaleAwareListener.inner" />
```

**Review by AdamKasp — APPROVED** [review]

File: features/admin/browsing_admin_panel.feature [hunk 2]
```diff
@@ -0,0 +1,24 @@
+@admin_locale
+Feature: Handling admin panel
```
Comment (GSadee) [inline features/admin/browsing_admin_panel.feature:2]:
```suggestion
Feature: Browsing administration panel in a proper locale
```
and adjust name of the file to this change

File: src/Sylius/Bundle/CoreBundle/EventListener/LocaleAwareListener.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    {
+        $this->decoratedListener->onKernelRequest($event);
+    }
+
+    public function onKernelFinishRequest(FinishRequestEvent $event): void
+    {
+        $this->decoratedListener->onKernelFinishRequest($event);
+    }
+
+    public static function getSubscribedEvents()
```
Comment (GSadee) [inline src/Sylius/Bundle/CoreBundle/EventListener/LocaleAwareListener.php:40]:
```suggestion
    public static function getSubscribedEvents(): array
```
?

File: src/Sylius/Bundle/LocaleBundle/Resources/config/services.xml [hunk 4]
```diff
@@ -51,7 +51,7 @@
         <service id="sylius.listener.request_locale_setter" class="Sylius\Bundle\LocaleBundle\Listener\RequestLocaleSetter">
             <argument type="service" id="sylius.context.locale" />
             <argument type="service" id="sylius.locale_provider" />
-            <tag name="kernel.event_listener" event="kernel.request" priority="4" />
+            <tag name="kernel.event_listener" event="kernel.request" priority="6" />
```
Comment (lchrusciel) [inline src/Sylius/Bundle/LocaleBundle/Resources/config/services.xml:54]: Can we check if we need to touch this place anyway? maybe we can just lower value here: https://github.com/Sylius/Sylius/pull/12458/files#diff-9511fd2b1a63d14d03a33b1037a4766e7305ed03adf9b848cbd3f56fa1e403d3R44 and leave this place as it was

Comment (SirDomin) [reply src/Sylius/Bundle/LocaleBundle/Resources/config/services.xml]: seems like its working well just with priority 4 here

**Review by arti0090 — APPROVED** [review]

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thanks, Łukasz! :tada:

**Pull request merged**